cli/src/utils/mockFs.ts

```diff
@@ -0,0 +1,45 @@
+import * as fs from 'fs';
+import path from 'path';
+import mockFs from 'mock-fs';
+
+export const getDirectoryAsObject = (directory: string, exclude = ['node_modules']) => {
+  const entries = fs.readdirSync(directory, { withFileTypes: true });
+  const result: Record<string, string> = {};
+
+  entries.forEach((entry) => {
+    if (!exclude.some((e) => entry.name.includes(e))) {
+      const entryPath = path.join(directory, entry.name);
+      if (entry.isFile()) {
+        result[entryPath] = fs.readFileSync(entryPath, 'utf8');
+      } else if (entry.isDirectory()) {
+        Object.assign(result, getDirectoryAsObject(entryPath, exclude));
+      }
+    }
+  });
+  return result;
+};
+
+const findDirectories = (baseDir: string, pattern: RegExp): string[] => {
+  const files = fs.readdirSync(baseDir);
+
+  const matchingDirs = files
+    .map((file) => path.join(baseDir, file))
+    .filter((filePath) => fs.statSync(filePath).isDirectory() && pattern.test(path.basename(filePath)));
+
+  return matchingDirs;
+};
+
+const NODE_MODULES_PATH = '../../../node_modules/.pnpm';
+const ABSOLUTE_MODULE_PATH = path.resolve(__dirname, NODE_MODULES_PATH);
+const COSMICONFIG_PATH = findDirectories(ABSOLUTE_MODULE_PATH, /cosmiconfig@.*/)[0];
+const OCLIF_PATH = findDirectories(ABSOLUTE_MODULE_PATH, /@oclif\+core@.+_@types\+node@.+_typescript@.+/)[0];
+
+export const mockFileSystem = (files: Record<string, string>) => {
```

Comment:
I tried very hard to use memfs but it doesn't work for node_modules which use `fs` internally. This solution seems to work well and runs fast. 

Here is how the code looked just as I decided to switch: https://github.com/xataio/client-ts/compare/main...xata-init-quality-improvements-2-3backup